Conduct code review with friends :p

MyLord916 opened the pull request:
File "/Users/mylord/PycharmProjects/aiogram_save_bot/file_managment/ya_file_manager.py", line 20, in upload_img
    await ya_disk_client.upload(path_to_file, dst_path)
TypeError: object SyncResourceLinkObject can't be used in 'await' expression

Вызываемая ошибка при всем этом не препятствует загрузке файлов, но ломает последующие действия по удалению сообщению

MyLord916 committed:
переписал функцию загрузки на асинхронный метод

StuBz211 commented on file_managment/ya_file_manager.py:
```diff
@@ -1,31 +1,31 @@
 
 load_dotenv(find_dotenv())
-y = yadisk.YaDisk(token=os.getenv('YA_TOKEN'))
+ya_disk_client = yadisk.YaDisk(token=os.getenv('YA_TOKEN'))
```
Тест комментарий

StuBz211 replied:
для асинхронной работы надо использовать асинхронный клиент.
```yadisk.AsyncClient(token=os.getenv('YA_TOKEN'))``` 


![image](https://github.com/user-attachments/assets/3dbceb9d-48e7-43f5-9cb8-f4fa51fea4a0)

https://pypi.org/project/yadisk/

MyLord916 replied:
![Снимок экрана 2024-08-15 в 12.06.38.png](https://github.com/user-attachments/assets/19949d2b-58cc-4089-9ff1-da0766449edf)

После оборота в функцию 
![Снимок экрана 2024-08-15 в 12.15.13.png](https://github.com/user-attachments/assets/620ea8c6-e889-428b-8722-8309c748ceef)

MyLord916 committed:
Сделал перестановку импортов согласно pep-8

MyLord916 committed:
Merge remote-tracking branch 'origin/develop' into fix

# Conflicts:
#	file_managment/ya_file_manager.py
#	handlers/user_handlers.py

MyLord916 committed:
Merge pull request #3 from MyLord916/fix

Сделал перестановку импортов согласно pep-8

MyLord916 committed:
rename file

MyLord916 committed:
Merge remote-tracking branch 'origin/develop' into develop

MyLord916 committed:
вынес утилиты, переписал клавиатуру и исправил импорты

MyLord916 commented:
Да и еще я откатился на синхронный метод загрузки пока не разберусь

StuBz211 commented on utils/utils.py:
```diff
@@ -0,0 +1,19 @@
+        subsequence += 1
+    else:
+        subsequence = 1
+    return dt_name
```
Как можно обойтись без global ?
предложу интересную штуку.
для subsequence использовать генератор-итератор.

```
def infinite_counter_gen():
    current = 0
    while True:
        if current > 100:
            current = 0
        yield current
        current += 1
        
 infinite_counter=infinite_counter_gen()
```
а далее ты используешь это в своей функции

```
dt_name =  datetime.now().strftime('%Y%m%d%H%M') + next(infinite_counter)
``

MyLord916 committed:
ya_file_manager.py
main() первые две строчки выдает ошибку закрытия клиента, последнюю не хочет проводить через await

MyLord916 committed:
поборов асинхронность мы уперлись в таймаут

MyLord916 committed:
Из aiohttp в httpx и обратно
Перебрал структуру модулей, подчистил коментарии и импорты

MyLord916 committed:
Исправил замечания
Убрал точку входа вya_file_manager.py
упаковал переменную со списком для хендлера клавиатуры в сам хендлер
запрятал set_correct_path в классе

MyLord916 committed:
add infinite current gen

MyLord916 committed:
Перенес фильтр директорий для хендлера в utils

MyLord916 committed:
recreated requirements.txt

MyLord916 committed:
Merge pull request #4 from MyLord916/fix

Fix
